src/transformers/data/datasets/language_modeling.py

```diff
@@ -109,3 +111,138 @@ def __len__(self):
 
     def __getitem__(self, i) -> torch.Tensor:
         return torch.tensor(self.examples[i], dtype=torch.long)
+
+class LineByLineWithSOPTextDataset(Dataset):
+    """
+    Dataset for sentence order prediction task, prepare sentence pairs for SOP task
+    """
+
+    def __init__(self, tokenizer: PreTrainedTokenizer, file_dir: str, block_size: int, random_seed: int):
```

Comment:
random seed is not used, could we remove?

Reply:
here we use `file_dir` which is a directory containing files. In other datasets I think we mostly use a single big file.

@sgugger should we modify this to only accept one big file, or should we keep it that way (introducing a discrepancy with other datasets).

Reply:
> random seed is not used, could we remove?

Hi here, random seed has been removed.

Reply:
> here we use `file_dir` which is a directory containing files. In other datasets I think we mostly use a single big file.
> 
> @sgugger should we modify this to only accept one big file, or should we keep it that way (introducing a discrepancy with other datasets).
there is error like
`__________________________________________________________________________________ DataCollatorIntegrationTest.test_sop_dataset_datacollator __________________________________________________________________________________
[gw0] darwin -- Python 3.7.6 /Users/amzliu/opt/anaconda3/bin/python

self = <tests.test_data_collator.DataCollatorIntegrationTest testMethod=test_sop_dataset_datacollator>

    def test_sop_dataset_datacollator(self):
        tokenizer = AutoTokenizer.from_pretrained('albert-base-v2')
        data_collator = DataCollatorForSOP(tokenizer)

>       dataset = LineByLineWithSOPTextDataset(tokenizer, file_dir=PATH_SAMPLE_TEXT_DIR, block_size=512, random_seed=42)

tests/test_data_collator.py:132:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/transformers/data/datasets/language_modeling.py:131: in __init__
    original_lines = f.readlines()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <encodings.utf_8.IncrementalDecoder object at 0x7fc035339990>
input = b'\x00\x00\x00\x01Bud1\x00\x00\x10\x00\x00\x00\x08\x00\x00\x00\x10\x00\x00\x00\x00\x86\x00\x00\x00\x00\x00\x00\x00\x00...00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
final = False

    def decode(self, input, final=False):
        # decode input (taking the buffer into account)
        data = self.buffer + input
>       (result, consumed) = self._buffer_decode(data, self.errors, final)
E       UnicodeDecodeError: 'utf-8' codec can't decode byte 0x86 in position 23: invalid start byte

../../opt/anaconda3/lib/python3.7/codecs.py:322: UnicodeDecodeError`

Could you help me figure this out? It works fine not using testing.

Reply:
@sgugger